src/Features/CSharp/Portable/Completion/CompletionProviders/AwaitCompletionProvider.cs

```diff
@@ -49,20 +52,77 @@ private protected override int GetSpanStart(SyntaxNode declaration)
 
         private protected override SyntaxNode? GetAsyncSupportingDeclaration(SyntaxToken token)
         {
-            var node = token.GetAncestor(node => node.IsAsyncSupportingFunctionSyntax());
-            // For local functions, make sure we either have arrow token or open brace token.
-            // Consider the following scenario:
-            // void Method()
-            // {
-            //     aw$$ AnotherMethodCall(); // NOTE: Here, the compiler produces LocalFunctionStatementSyntax.
-            // }
-            // For this case, we're interested in putting async in front of Method()
-            if (node is LocalFunctionStatementSyntax { ExpressionBody: null, Body: null })
+            var declaration = token.GetAncestor(node => node.IsAsyncSupportingFunctionSyntax());
+            // In a case like
+            //   someTask.$$
+            //   await Test();
+            // someTask.await Test() is parsed as a local function statement.
+            // We skip this and look further up in the hierarchy.
+            if (declaration is LocalFunctionStatementSyntax localFunction && localFunction.ReturnType.ChildTokens().Contains(token))
```

Comment:
doing things like .ChildTokens is too vague.  we should enumerate the exact cases we expect with the exact node structure that causes this (i.e. the token being a dot that it's a qualified name).

Reply:
Changed in cabd453:

* Placed the check before the common case
* `token.Parent` must be `QualifiedNameSyntax`
* Be explicit about the expected tree structure without any "Ancestors" logic.